tests/Aspire.Hosting.Azure.Tests/AzureUserAssignedIdentityTests.cs

```diff
@@ -99,4 +99,179 @@ public async Task AddAzureUserAssignedIdentity_WithRoleAssignments_Works()
+        Assert.Collection(model.Resources,
+            r => Assert.IsType<AzureEnvironmentResource>(r),
+            r => Assert.IsType<AzureContainerAppEnvironmentResource>(r),
+            r => Assert.IsType<AzureStorageResource>(r),
+            r => Assert.IsType<AzureUserAssignedIdentityResource>(r),
+            r => Assert.IsType<ProjectResource>(r),
+            r => Assert.IsType<ProjectResource>(r),
+            r => Assert.True(r is AzureProvisioningResource { Name: "myapp-roles-mystorage" }),
+            r => Assert.True(r is AzureProvisioningResource { Name: "myapp2-roles-mystorage" }));
+
+        // Verify the identity resource is the only one that exists
+        var identityResource = Assert.Single(model.Resources.OfType<AzureUserAssignedIdentityResource>());
+        Assert.Equal("myidentity", identityResource.Name);
+
+        // Verify that both compute resources have the same identity annotation
+        var computeResource = Assert.Single(model.Resources.OfType<IComputeResource>(), r => r.Name == "myapp");
+        var identityAnnotation = Assert.Single(computeResource.Annotations.OfType<AppIdentityAnnotation>());
+        var computeResource2 = Assert.Single(model.Resources.OfType<IComputeResource>(), r => r.Name == "myapp2");
+        var identityAnnotation2 = Assert.Single(computeResource2.Annotations.OfType<AppIdentityAnnotation>());
+        Assert.Same(identity.Resource, identityAnnotation.IdentityResource);
+        Assert.Same(identity.Resource, identityAnnotation2.IdentityResource);
+
+        // Verify the role assignment resource for the project
+        var roleAssignmentResource = Assert.Single(model.Resources.OfType<AzureProvisioningResource>(),
+            r => r.Name == "myapp-roles-mystorage");
+        var roleAssignmentResource2 = Assert.Single(model.Resources.OfType<AzureProvisioningResource>(),
+            r => r.Name == "myapp2-roles-mystorage");
+        // Each project uses the same identity, but assigns different roles
+        Assert.NotSame(roleAssignmentResource, roleAssignmentResource2);
+
+        // Get the Bicep for all resources
+        var deploymentTarget = Assert.Single(computeResource.Annotations.OfType<DeploymentTargetAnnotation>());
+        var deploymentTarget2 = Assert.Single(computeResource2.Annotations.OfType<DeploymentTargetAnnotation>());
+        var (_, computeResourceBicep) = await GetManifestWithBicep(deploymentTarget.DeploymentTarget, skipPreparer: true);
+        var (_, computeResourceBicep2) = await GetManifestWithBicep(deploymentTarget2.DeploymentTarget, skipPreparer: true);
+        var (_, identityBicep) = await GetManifestWithBicep(identityResource, skipPreparer: true);
+        var (_, roleBicep) = await GetManifestWithBicep(roleAssignmentResource, skipPreparer: true);
+        var (_, roleBicep2) = await GetManifestWithBicep(roleAssignmentResource2, skipPreparer: true);
+
+        Target[] targets = [
```

Comment:
Hmm is this what we should be doing everywhere else for multiple files?

Reply:
There's also this API for multi-file verification:

```csharp
await Verify(envManifest.ToString(), "json")
              .AppendContentAsFile(envBicep, "bicep")
              .AppendContentAsFile(containerManifest.ToString(), "json")
              .AppendContentAsFile(containerBicep, "bicep")
              .AppendContentAsFile(registryManifest.ToString(), "json")
              .AppendContentAsFile(registryBicep, "bicep");
```

I prefer this style because I think it's a bit easier to read than multiple chained calls but we use the `AppendContentAsFile` extension method more frequently in the codebase so I'll conform with the masses.

Reply:
Can you use the same style as the others and then do pass to fix them all?